extensions/amp-form/0.1/amp-form.js

```diff
@@ -242,22 +247,29 @@ export class AmpForm {
           xhrUrl = this.xhrAction_;
           body = new FormData(this.form_);
         }
-        return this.xhr_.fetchJson(dev().assertString(xhrUrl), {
+        return this.xhr_.fetch(dev().assertString(xhrUrl), {
           body,
           method: this.method_,
           credentials: 'include',
         }).then(response => {
-          this.triggerAction_(/* success */ true, response);
-          // TODO(mkhatib, #6032): Update docs to reflect analytics events.
-          this.analyticsEvent_('amp-form-submit-success');
-          this.setState_(FormState_.SUBMIT_SUCCESS);
-          this.renderTemplate_(response || {});
-        }).catch(error => {
+          return response.json().then(json => {
+            this.triggerAction_(/* success */ true, json);
+            this.analyticsEvent_('amp-form-submit-success');
+            this.setState_(FormState_.SUBMIT_SUCCESS);
+            this.renderTemplate_(json || {});
+            this.maybeHandleRedirect_(
+                /** @type {../../../src/service/xhr-impl.FetchResponse} */ (
+                    response));
+          });
+        }, error => {
```

Comment:
This needs to be in a `#catch`, so it can handle errors caused by the above nested JSON parse.